internal/factsengine/gatherers/sbddump.go

```diff
@@ -0,0 +1,129 @@
+package gatherers
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/internal/core/cluster"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+	"github.com/trento-project/agent/pkg/utils"
+)
+
+const (
+	SBDDumpGathererName = "sbd_dump"
+)
+
+var undesiredParenthesesRegexp = regexp.MustCompile(`[()]`)
+
+// nolint:gochecknoglobals
+var (
+	SBDDevicesLoadingError = entities.FactGatheringError{
+		Type:    "sbd-devices-loading-error",
+		Message: "error loading the configured sbd devices",
+	}
+
+	SBDDumpCommandError = entities.FactGatheringError{
+		Type:    "sbd-dump-command-error",
+		Message: "error while executing sbd dump",
+	}
+)
+
+type SBDDumpGatherer struct {
+	executor      utils.CommandExecutor
+	sbdConfigFile string
+}
+
+func NewDefaultSBDDumpGatherer() *SBDDumpGatherer {
+	return NewSBDDumpGatherer(utils.Executor{}, cluster.SBDConfigPath)
+}
+
+func NewSBDDumpGatherer(executor utils.CommandExecutor, sbdConfigFile string) *SBDDumpGatherer {
+	return &SBDDumpGatherer{
+		executor:      executor,
+		sbdConfigFile: sbdConfigFile,
+	}
+}
+
+func (gatherer *SBDDumpGatherer) Gather(factsRequests []entities.FactRequest) ([]entities.Fact, error) {
+	facts := []entities.Fact{}
+	log.Infof("Starting %s facts gathering process", SBDDumpGathererName)
+
+	configuredDevices, err := loadDevices(gatherer.sbdConfigFile)
+
+	if err != nil {
+		return facts, SBDDevicesLoadingError.Wrap(err.Error())
+	}
+
+	for _, factRequest := range factsRequests {
+		var fact entities.Fact
+
+		if devicesDumps, err := getSBDDevicesDumps(gatherer.executor, configuredDevices); err == nil {
+			fact = entities.NewFactGatheredWithRequest(factRequest, devicesDumps)
+		} else {
+			gatheringError := SBDDumpCommandError.Wrap(err.Error())
+
+			fact = entities.NewFactGatheredWithError(factRequest, gatheringError)
+		}
+		facts = append(facts, fact)
+	}
+
+	log.Infof("Requested %s facts gathered", SBDDumpGathererName)
+	return facts, nil
+}
+
+func loadDevices(sbdConfigFile string) ([]string, error) {
+	conf, err := cluster.LoadSbdConfig(sbdConfigFile)
+
+	if err != nil {
+		return nil, err
+	}
+
+	configuredDevices, found := conf["SBD_DEVICE"]
```

Comment:
One things here. The SBD_DEVICE value can come between quotes:
```
SBD_DEVICE="/dev/vdc;/dev/vdb"
```
Are we handling this properly. I know we do in the main sbd code code, but I don't know if the `LoadSbdConfig` does the magic.
We could include a test case for this

Reply:
I tested manually changing the fixture file to both with/without quotes and works.

If you think the extra test is really needed, it could be a candidate for another PR :sweat_smile:

Reply:
Why other PR? Come on man! XD

Reply:
:stuck_out_tongue_closed_eyes: ok